homeassistant/components/nws/__init__.py

```diff
@@ -58,36 +60,46 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     nws_data = SimpleNWS(latitude, longitude, api_key, client_session)
     await nws_data.set_station(station)
 
-    async def update_observation() -> None:
-        """Retrieve recent observations."""
-        await call_with_retry(
+    def async_setup_update_observation(
+        retry_interval: datetime.timedelta | float,
+        retry_stop: datetime.timedelta | float,
+    ) -> Callable[[], Awaitable[None]]:
+        return partial(
+            call_with_retry,
             nws_data.update_observation,
-            RETRY_INTERVAL,
-            RETRY_STOP,
+            retry_interval,
+            retry_stop,
             start_time=utcnow() - UPDATE_TIME_PERIOD,
```

Comment:
I'm not sure this one will work because you want `utcnow` to be called at the time its run instead of when the partial is created

Reply:
Sorry for my sloppy application of your precise patch.